### PR TITLE
[Core] Transition to fix #7052

### LINKED ISSUE #7052: Normal computation on condition using CalculateNormal3D for 4 noded (quad)
Hi All, I was wondering if the functionality of the CalculateNormal3D in NormalCalculationUtils could be extended to linear quads. Currently, it considers only 3 nodes in the normal computation.

Any help/thought much appreciated, thank you.

-----------------------------------------------
@ddiezrod 
@pooyan-dadvand

### PR DESCRIPTION
**Pull request by loumalouomega**
Separating header and source in CalculateOnSimplex (I corrected also tabulations and {} position to the style)

Adding test (non-existing until now)

Next step will be to create the CalculateNormals suggested by @pooyan-dadvand which will replace the MortarUtilities version (I will create two version, one unit normal and the other as area normal, right now I only consider unit nornmal in MortarUtilities)

Refers to #7052

### CODE REVIEW

File: kratos/utilities/normal_calculation_utils.h [hunk 1]
```diff
@@ -11,257 +11,146 @@
...
-
-/// Tool to evaluate the normals on nodes based on the normals of a set of surface conditions
+/**
+ * @namespace NormalCalculationUtils
+ * @ingroup KratosCore
+ * @brief Tool to evaluate the normals on nodes based on the normals of a set of surface conditions
+ * @author Pooyan Dadvand
+ * @author Riccardo Rossi
+ */
 class NormalCalculationUtils
```
Comment (ddiezrod) [inline kratos/utilities/normal_calculation_utils.h]: I think you need the KRATOS_API here

Comment (loumalouomega) [reply kratos/utilities/normal_calculation_utils.h]: Yep

**Comment by loumalouomega** [comment]
@roigcarlo several PR are now failing because: 

´MSBUILD : error MSB1009: Project file does not exist.´

**Comment by loumalouomega** [comment]
Ping @ddiezrod and I will add the extension for any geometry in a future PR

**Comment by loumalouomega** [comment]
Refering #7052

**Review by ddiezrod — APPROVED** [review]
Thanks @loumalouomega !

**Pull request merged**